skyportal/models.py

```diff
@@ -718,99 +718,107 @@ class Obj(Base, ha.Point):
         doc="Notifications regarding the object sent out by users",
     )
 
-    @hybrid_property
-    def last_detected_at(self):
+    @hybrid_method
+    def last_detected_at(self, user):
         """UTC ISO date at which the object was last detected above a given S/N (3.0 by default)."""
         detections = [
             phot.iso
-            for phot in self.photometry
+            for phot in Photometry.query_records_accessible_by(user)
+            .filter(Photometry.obj_id == self.id)
+            .all()
             if phot.snr is not None and phot.snr > PHOT_DETECTION_THRESHOLD
         ]
         return max(detections) if detections else None
 
     @last_detected_at.expression
-    def last_detected_at(cls):
+    def last_detected_at(cls, user):
         """UTC ISO date at which the object was last detected above a given S/N (3.0 by default)."""
         return (
-            sa.select([sa.func.max(Photometry.iso)])
-            .where(Photometry.obj_id == cls.id)
-            .where(Photometry.snr.isnot(None))
-            .where(Photometry.snr > PHOT_DETECTION_THRESHOLD)
+            Photometry.query_records_accessible_by(
+                user, columns=[sa.func.max(Photometry.iso)], mode="read"
+            )
+            .filter(Photometry.obj_id == cls.id)
+            .filter(Photometry.snr.isnot(None))
+            .filter(Photometry.snr > PHOT_DETECTION_THRESHOLD)
```

Comment:
nicely done

Reply:
bet this translates to a pretty complex sql query under the hood! this is a good illustration of the expressive power of the ORM